client/app/services/blueprints-state.service.js

```diff
@@ -78,6 +78,57 @@
       return blueprint.doNotSaveFlag;
     };
 
+    blueprint.publishBlueprint = function(tmpBlueprint) {
+      var deferred = $q.defer();
+      var origBlueprint = blueprint.getOriginalBlueprint();
+      if (!angular.equals(origBlueprint, tmpBlueprint)) {
+        // save the changed blueprint
+        blueprint.saveBlueprint(tmpBlueprint).then(function(id) {
+          blueprint.publishBlueprintOp(tmpBlueprint).then(function() {
+            deferred.resolve();
+          }, publishfailure);
+        }, savefailure);
+      } else {
+        // just publish the unchanged blueprint
+        blueprint.publishBlueprintOp(tmpBlueprint).then(function() {
```

Comment:
Action not required: consider using a named function `success` and subsequent `failure`

(really at the end of the day this removes the thinking from naming convention, something i hate to spend time on "oooooh what should i name *THIS*)